test/rut_spec.js

```diff
@@ -2,113 +2,153 @@
+
+      scope = $rootScope.$new();
       scope.inputs = { rut: '' };
       $compile(element)(scope);
       scope.$digest();
+
       element = element.find('input');
     }));
 
-    it('should pass with valid rut', function() {
-      scope.form.rut.$setViewValue('99.999.999-9');
-      expect(scope.form.rut.$valid).toEqual(true);
+    it('should not format the rut in real time', function() {
+      setInputValue(element, '999999999');
+      expect(element.val()).toEqual('999999999');
     });
 
-    it('should not pass with invalid rut', function() {
-      scope.form.rut.$setViewValue('1.018.177-6');
-      expect(scope.form.rut.$valid).toEqual(false);
-    });
+    it('should not format the rut in real time', function() {
+      setInputValue(element, '999999999');
+
+      element.triggerHandler('blur');
 
-    // TODO: validate on blur/focus behavior.
+      expect(element.val()).toEqual('99.999.999-9');
+    });
 
   });
 
-  describe('rut angular-validate validator', function() {
+  describe('rut filter', function(){
+    var rutFilter;
 
-    var element, scope;
+    beforeEach(inject(function($filter) {
+      rutFilter = $filter('rut');
+    }));
 
-    beforeEach(inject(function($rootScope, $compile) {
-      element = angular.element('<form name="form"><input name="rut" type="text" ng-model="inputs.rut" validate="rut"/></form>');
-      scope = $rootScope;
-      scope.inputs = { rut: '' };
-      $compile(element)(scope);
-      scope.$digest();
-      element = element.find('input');
+    it('should properly format a rut', function() {
+      expect(rutFilter('1')).toEqual('1');
+      expect(rutFilter('101818k')).toEqual('101.818-K');
+      expect(rutFilter('101818l')).toEqual('10.181-8');
+      expect(rutFilter('1018189982')).toEqual('101.818.998-2');
+    });
+
+    it('should return default value if no rut is given', function() {
+      expect(rutFilter('', '-')).toEqual('-');
+      expect(rutFilter(null, 'NA')).toEqual('NA');
+      expect(rutFilter(200, 'NA')).toEqual('NA');
+    });
+  });
+
+  describe('rutHelper', function(){
+    var rutHelper;
+
+    beforeEach(inject(function(RutHelper) {
+      rutHelper = RutHelper;
     }));
 
-    it('should pass with valid rut', function() {
-      scope.form.rut.$setViewValue('99.999.999-9');
-      expect(scope.form.rut.$valid).toEqual(true);
+    it('should properly format a rut', function() {
+      expect(rutHelper.format('1')).toEqual('1');
+      expect(rutHelper.format('101818k')).toEqual('101.818-K');
+      expect(rutHelper.format('101818l')).toEqual('10.181-8');
+      expect(rutHelper.format('1018189982')).toEqual('101.818.998-2');
     });
 
-    it('should not pass with invalid rut', function() {
-      scope.form.rut.$setViewValue('1.018.177-6');
-      expect(scope.form.rut.$valid).toEqual(false);
+    it('should properly validate a rut', function() {
```

Comment:
'it' is not defined                                           no-undef<br>Unexpected function expression                                prefer-arrow-callback<br>Missing space before function parentheses                     space-before-function-paren